src/System.Management.Automation/engine/SessionStateLocationAPIs.cs

```diff
@@ -262,10 +263,18 @@ internal PathInfo SetLocation(string path, CmdletProviderContext context)
                     // Since the path is an absolute path
                     // we need to change the current working
                     // drive
-
                     PSDriveInfo newWorkingDrive = GetDrive(driveName);
                     CurrentDrive = newWorkingDrive;
 
+                    // If the path is simply a colon-terminated drive,
+                    // not a slash-terminated path to the root of a drive,
+                    // set the path to the current working directory of that drive.
+                    string colonTerminatedVolume = CurrentDrive.Name + ':';
+                    if (CurrentDrive.VolumeSeparatedByColon && (path.Length == colonTerminatedVolume.Length))
+                    {
+                        path = Path.Combine((colonTerminatedVolume + "\\"), CurrentDrive.CurrentLocation);
```

Comment:
> colonTerminatedVolume + "\\"

Please use `Path.DirectorySeparatorChar` instead of `"\\"`. I will make that change for you.

Reply:
This has been changed.